### PR TITLE
Use HandlerContext pattern for go dependency injection

### PR DESCRIPTION
**Pull request by macrael**
## Description

This change updates work @jim and I did to make code dependencies explicit for all of our handlers. It adds an ADR that explains why we are using explicit dependencies and the strategy for making that code a bit less redundant right now. There is code in this PR that updates a few other handlers to have explicit dependencies as well as code to refactor the implementation of the existing handlers to share a base struct (HandlerContext) that each individual handler is merely a type alias for. This way we don't have to litter our code with the same two dependencies and can more easily update the dependencies for all the handlers at the same time 

## Verification Steps

* [ ] All tests pass.
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely) have been satisfied.

## References

* [Pivotal #155788306](https://www.pivotaltracker.com/story/show/155788306)
* [Parse: Dependency Injection with Go](http://blog.parseplatform.org/learn/engineering/dependency-injection-with-go/) explains more about the approach used.
* [This example go-swagger app](https://github.com/go-openapi/kvstore) shows a similar approach with an app-wide context that dependencies are put into.

### CODE REVIEW

File: swagger/api.yaml [hunk 1]
```diff
@@ -19,6 +19,8 @@ paths:
       summary: List all TSPs
       description: Gets a list of all TSPs registered.
       operationId: indexTSPs
+      tags:
+        - tsps
       x-access: Access to this endpoint is restricted to members of the Admin, Transcom & JPPSO user groups
```
Comment (ntwyman) [inline swagger/api.yaml]: Why are these changes here? Doesn't seem to have anything to do with initialization code.

Comment (macrael) [reply swagger/api.yaml]: Adding a tag puts all the associated generated parameters, responses, and handlers into their own package under pkg/gen/restapi/apioperations. We are doing this for all the internal apis so I thought it would be appropriate to normalize them while I was normalizing the initializers. The change this led to in our code was the separate imports for shipment and tsp operations in handlers/tsps.go. 

I'm happy to pull this out into it's own PR if you'd rather consider it that way.
